model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_edge_diagnostics_for_velocity_advection.py

```diff
@@ -6,151 +6,48 @@
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
 import gt4py.next as gtx
-from gt4py.next.ffront.fbuiltins import broadcast, where
+from gt4py.next.ffront.fbuiltins import astype, where
 
 from icon4py.model.atmosphere.dycore.stencils.compute_contravariant_correction import (
     _compute_contravariant_correction,
 )
 from icon4py.model.atmosphere.dycore.stencils.compute_horizontal_advection_term_for_vertical_velocity import (
     _compute_horizontal_advection_term_for_vertical_velocity,
 )
-from icon4py.model.atmosphere.dycore.stencils.compute_horizontal_kinetic_energy import (
-    _compute_horizontal_kinetic_energy,
-)
 from icon4py.model.atmosphere.dycore.stencils.compute_tangential_wind import (
     _compute_tangential_wind,
 )
 from icon4py.model.atmosphere.dycore.stencils.extrapolate_at_top import _extrapolate_at_top
-from icon4py.model.atmosphere.dycore.stencils.interpolate_vn_to_half_levels_and_compute_kinetic_energy_on_edges import (
-    _interpolate_vn_to_half_levels_and_compute_kinetic_energy_on_edges,
-)
 from icon4py.model.atmosphere.dycore.stencils.mo_icon_interpolation_scalar_cells2verts_scalar_ri_dsl import (
     _mo_icon_interpolation_scalar_cells2verts_scalar_ri_dsl,
 )
 from icon4py.model.common import dimension as dims, field_type_aliases as fa, type_alias as ta
-from icon4py.model.common.interpolation.stencils.interpolate_edge_field_to_half_levels_vp import (
-    _interpolate_edge_field_to_half_levels_vp,
-)
+from icon4py.model.common.dimension import Koff
+from icon4py.model.common.type_alias import vpfloat, wpfloat
 
 
 @gtx.field_operator
-def _compute_vt_vn_on_half_levels_and_kinetic_energy(
-    vn: fa.EdgeKField[ta.wpfloat],
-    wgtfac_e: fa.EdgeKField[ta.vpfloat],
-    tangential_wind_on_half_levels: fa.EdgeKField[ta.wpfloat],
-    tangential_wind: fa.EdgeKField[ta.vpfloat],
-    vn_on_half_levels: fa.EdgeKField[ta.vpfloat],
-    horizontal_kinetic_energy_at_edges_on_model_levels: fa.EdgeKField[ta.vpfloat],
+def _interpolate_to_half_levels(
```

Comment:
We arleady have `interpolate_to_half_levels` field operator somewhere else, need to unify.